components/coding/studentPortal/Sidebar.tsx

```diff
@@ -51,6 +54,19 @@ export const Sidebar: React.FC<SidebarProps> = ({}: SidebarProps) => {
     }
   }, [router.pathname]);
 
+  const { data } = useQuery(FETCH_USER_GOALS, {
```

Comment:
i think a better approach would be to do the aggregation on the GQL side... i.e create a new query like FETCH_UPCOMING_GOALS that sums the number of goals due within the following week for that user